tests/e2e/Services/VCS/VCSConsoleClientTest.php

```diff
@@ -0,0 +1,316 @@
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId,
+            'providerRepositoryId' => $this->providerRepositoryId
+        ]);
+
+        $this->assertEquals(200, $runtime['headers']['status-code']);
+        $this->assertEquals($runtime['body']['runtime'], 'ruby-3.1');
+
+        /**
+         * Test for FAILURE
+         */
+
+        $runtime = $this->client->call(Client::METHOD_POST, '/vcs/github/installations/' . $installationId . '/providerRepositories/randomRepositoryId/detection', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId,
+            'providerRepositoryId' => 'randomRepositoryId'
+        ]);
+
+        $this->assertEquals(404, $runtime['headers']['status-code']);
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testListRepositories(string $installationId)
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        $repositories = $this->client->call(Client::METHOD_GET, '/vcs/github/installations/' . $installationId . '/providerRepositories', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId
+        ]);
+
+        $this->assertEquals(200, $repositories['headers']['status-code']);
+        $this->assertEquals($repositories['body']['total'], 3);
+
+        $searchedRepositories = $this->client->call(Client::METHOD_GET, '/vcs/github/installations/' . $installationId . '/providerRepositories', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId,
+            'search' => 'func'
+        ]);
+
+        $this->assertEquals(200, $searchedRepositories['headers']['status-code']);
+        $this->assertEquals($searchedRepositories['body']['total'], 1);
+
+        /**
+         * Test for FAILURE
+         */
+
+        $repositories = $this->client->call(Client::METHOD_GET, '/vcs/github/installations/randomInstallationId/providerRepositories', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => 'randomInstallationId'
+        ]);
+
+        $this->assertEquals(404, $repositories['headers']['status-code']);
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testGetRepository(string $installationId, string $providerRepositoryId2 = '705764449')
```

Comment:
`providerRepositoryId2` should probably be class variable like `providerRepositoryId`. So its at the top of the file, easy to find and change if needed.